src/backend/joanie/core/models/products.py

```diff
@@ -437,23 +437,50 @@ def main_invoice(self):
 
     def clean(self):
         """Clean instance fields and raise a ValidationError in case of issue."""
+        course_product_relation = courses_models.CourseProductRelation.objects.filter(
+            course=self.course_id,
+            product=self.product_id,
+            organizations=self.organization_id,
+        )
```

Comment:
In order to save one query, we could use `first` directly.

```suggestion
        course_product_relation = courses_models.CourseProductRelation.objects.filter(
            course=self.course_id,
            product=self.product_id,
            organizations=self.organization_id,
        ).only('max_validated_orders').first()
```